src/lib/integration_tests/tests/upgrade/feed_filter_upgrade_test.rs

```diff
@@ -0,0 +1,431 @@
+use std::{collections::HashMap, thread, time::Duration};
+
+use candid::{encode_one, CandidType, Deserialize};
+use pocket_ic::{PocketIc, WasmResult};
+use shared_utils::{
+    canister_specific::{
+        individual_user_template::types::{
+            arg::IndividualUserTemplateInitArgs, post::PostDetailsFromFrontend,
+        },
+        post_cache::types::arg::PostCacheInitArgs,
+    },
+    common::types::{
+        known_principal::{KnownPrincipalMap, KnownPrincipalType},
+        top_posts::post_score_index_item::{PostScoreIndexItem, PostScoreIndexItemV1},
+    },
+    types::canister_specific::post_cache::error_types::TopPostsFetchError,
+};
+use test_utils::setup::test_constants::{
+    get_mock_user_alice_principal_id, get_mock_user_bob_principal_id,
+};
+
+const OLD_POST_CACHE_WASM_PATH: &str =
+    "../../../target/wasm32-unknown-unknown/release/old_post_cache.wasm.gz";
+const OLD_INDIVIDUAL_TEMPLATE_WASM_PATH: &str =
+    "../../../target/wasm32-unknown-unknown/release/old_individual_user_template.wasm.gz";
```

Comment:
I am not sure about this test. Do we need to manually build `old_post_cache.wasm.gz` and `old_individual_user_template_wasm.gz`. Also I think this test is related to changed we would be adding for this upgrade. Will the test fail in subsequent changes?

Reply:
It was more for my sanity testing since I didn't know about ic-repl-tests before. Yes, it would be failing in the subsequent changes. Need to remove post this. Let me add this under a different feature for now so that it won't run in `cargo test`. Can remove it later